blog.qmd

```diff
@@ -0,0 +1,228 @@
+from Florida's community colleges using web scraping techniques. We successfully
+compiled a comprehensive dataset, but the question remained: how do we make
+sense of this vast amount of text data? Enter quanteda, an R package designed
+for quantitative text analysis.
+
+## Getting Started with Quanteda
+
+Quanteda, short for Quantitative Analysis of Textual Data, is a powerful tool
+for managing and analyzing text data in R. It offers a suite of functions for corpus management, creating document-feature matrices, analyzing keywords, and more. These functions are highly efficient and provide a consistent interface with support for multiple languages. While it operates as a standalone package, it also integrates seamlessly with extensions such as readtext, spacyr, and quanteda.textstats.
+
+To install and load the required packages, we can use the librarian package for convenience
+```{r text-}
+librarian::shelf(tidyverse, # data wrangling
+                 quanteda, # text mining
+                 readtext # read texts and associated document-level meta-data
+)
+```
+
+Next, we load our text data containing course descriptions as well as document-level meta-data using the readtext::readtext() function. We then perform some additional cleaning steps to standardize variable names and focus our analysis on active courses for credit.
+
+```{r}
+courses <-
+  readtext(here::here("data/data-intermediate/course-descriptions.csv"), 
+           text_field = "course_description",
+           docid_field = 'course_id') %>% 
+  janitor::clean_names() %>% 
+  filter(course_status == "ACTIVE",
+         type_of_credit == "COLLEGE CREDIT")
+```
+
+```{r}
+#| echo: false
+courses %>% 
+  slice(1:1000) %>% 
+  gt() %>% 
+  opt_interactive(use_text_wrapping = FALSE, use_resizers = TRUE, use_compact_mode = TRUE) %>% 
+  my_theme()
+```
+
+The first step in our analysis is to create a corpus, a collection of text documents, from our course descriptions. We can achieve this using the corpus() function in quanteda. Next, we extract the tokens in the corpus—usually words, but they can also be n-grams or multi-word expressions. The tokens function allows us to define what we mean by tokens and apply some rules to ignore elements such as punctuation and digits.
+
+```{r}
+corpus <- corpus(courses)
+```
+
+```{r}
+#| echo: false
+corpus[1:5]
+```
+
+```{r}
+tk <- tokens(corpus, what = "word", remove_punct = TRUE, remove_numbers = TRUE)
+```
+
+```{r}
+#| echo: false
+tk[1:5]
+```
+
+## Key-Term Searches with a Dictionary
+
+Our primary interest lies in identifying courses related to different types of work-based learning such as internships, apprenticeships, or practicums. For each type of work-based learning experience, we create a list of terms that we want to treat equivalently. For instance, our dictionary can specify that a course description refers to a clinical WBL if either of the terms "clinicals" or "clinical experience" appear.
```

Comment:
This is helpful background! could inform that paragraph in the Intro that I commented is missing

Reply:
@alenastern